### PR TITLE
Linsun/convint8 fwd instances

### PR DESCRIPTION
**Pull request by linsun12**
This pull request is for adding int8 support for 2d grouped convolution fwd on gfx9 architecture using xdlops. The following instances are added:

~~For layout GNHWC/GKYXC/GNHWK~~
~~grouped_conv2d_fwd_xdl_gnhwc_gkyxc_gnhwk_int8_instances~~

For layout NHWGC/GKYXC/NHWGK
grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_instances
grouped_conv2d_fwd_xdl_large_tensor_nhwgc_gkyxc_nhwgk_int8_instances
grouped_conv2d_fwd_xdl_merged_groups_nhwgc_gkyxc_nhwgk_int8_instances
grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_comp_instances
grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_mem_intra_instances
grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_mem_inter_instances

For layout NGCHW/GKYXC/NGKHW
grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_instances
grouped_conv2d_fwd_xdl_merged_groups_ngchw_gkyxc_ngkhw_int8_instances
grouped_conv2d_fwd_xdl_ngchw_gkyxc_ngkhw_int8_comp_instances
grouped_conv2d_fwd_xdl_ngchw_gkyxc_ngkhw_int8_mem_intra_instances
grouped_conv2d_fwd_xdl_ngchw_gkyxc_ngkhw_int8_mem_inter_instances

### CODE REVIEW

**Comment by carlushuang** [comment]
Hi @bartekxk can you help review this? looks like new instances for group-conv

**Comment by carlushuang** [comment]
@linsun12 can you elaborate more on what quantization/dequant used in this int8 conv instances? looks like you just compute the conv as is it?

**Comment by bartekxk** [comment]
Hi @carlushuang , I will review this change on monday

File: .gitignore [hunk 1]
```diff
@@ -35,6 +35,7 @@ tags
 
 # Editors
 .vscode
+CMakePresets.json
```
Comment (bartekxk) [inline .gitignore]: Why we need this?

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp [hunk 2]
```diff
@@ -189,9 +192,17 @@ struct DeviceOperationInstanceFactory<ck::tensor_operation::device::DeviceGroupe
             {
                 add_device_grouped_conv2d_fwd_xdl_gnhwc_gkyxc_gnhwk_bf16_instances(op_ptrs);
             }
+#endif
+#ifdef CK_ENABLE_INT8
```
Comment (bartekxk) [inline library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp]: Hi, I asked for remove gnhwc instances in previous PR.

File: library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/comp/device_grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_comp_instance.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/comp/device_grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_comp_instance.cpp]: Here and bellow:
```suggestion
// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
```

File: library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/merged_groups/device_grouped_conv2d_fwd_xdl_merged_groups_nhwgc_gkyxc_nhwgk_int8_instance.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    add_device_operation_instances(
+        instances,
+        device_grouped_conv_fwd_xdl_merged_groups_int8_instances<2,
+                                                                 NHWGC,
+                                                                 GKYXC,
+                                                                 Empty_Tuple,
+                                                                 NHWGK,
+                                                                 ConvFwd3x3>{});
+}
+
```
Comment (bartekxk) [inline library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/merged_groups/device_grouped_conv2d_fwd_xdl_merged_groups_nhwgc_gkyxc_nhwgk_int8_instance.cpp:44]: Random place: Could you also update profile_grouped_conv_fwd.cpp and tests?

**Review by bartekxk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (linsun12) [reply .gitignore]: This is the vscode config file I use for cmake

**Comment by linsun12** [comment]
> @linsun12 can you elaborate more on what quantization/dequant used in this int8 conv instances? looks like you just compute the conv as is it?

Hi Carlus there's no quantization/dequantization used in these instances, their input and output datatypes are int8.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp (hunk 2 above)
Comment (linsun12) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp]: I sent you an msg earlier explaining why this was not deleted, but if there's no use case for this instance, I will delete it later. Will create another PR.

File: library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/comp/device_grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_comp_instance.cpp (hunk 3 above)
Comment (linsun12) [reply library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/comp/device_grouped_conv2d_fwd_xdl_nhwgc_gkyxc_nhwgk_int8_comp_instance.cpp]: will change in new PR.

File: .gitignore (hunk 1 above)
Comment (linsun12) [reply .gitignore]: removed in latest commit

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp (hunk 2 above)
Comment (linsun12) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_forward.hpp]: deleted in latest commit

File: library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/merged_groups/device_grouped_conv2d_fwd_xdl_merged_groups_nhwgc_gkyxc_nhwgk_int8_instance.cpp (hunk 4 above)
Comment (linsun12) [reply library/src/tensor_operation_instance/gpu/grouped_conv2d_fwd/xdl/merged_groups/device_grouped_conv2d_fwd_xdl_merged_groups_nhwgc_gkyxc_nhwgk_int8_instance.cpp:44]: test_grouped_convnd_fwd.cpp already defined the test cases for grouped conv 2d int8. Removed GNHWC/GKYXC/GNHWK, added NGCHW/GKYXC/NGKHW, leaving NHWGC/GKYXC/NHWGK. NHWGC/GKYXC/NHWGK is added for large tensor unit tests within test_grouped_convnd_fwd_large_cases_xdl.cpp.

**Review by bartekxk — APPROVED** [review]
LGTM

**Comment by linsun12** [comment]
image

**Pull request merged**